static/js/actions/course_enrollments.js

```diff
@@ -22,9 +23,10 @@ export const addCourseEnrollment = (courseId: string): Dispatcher<*> => {
     dispatch(requestAddCourseEnrollment(courseId));
     return api.addCourseEnrollment(courseId).
       then(() => {
-        dispatch(receiveAddCourseEnrollmentSuccess());
-        dispatch(fetchDashboard(SETTINGS.user.username));
-        dispatch(fetchCoursePrices());
+        dispatch(receiveAddCourseEnrollmentSuccess(courseId));
+        dispatch(fetchDashboard(SETTINGS.user.username, true));
+        dispatch(fetchCoursePrices(true));
+        dispatch(showEnrollPayLaterSuccessMessage());
```

Comment:
probably you want the `.then` callback here to return this call (`dispatch(showEnrollPayLaterSuccessMessage());`)